### PR TITLE
LIME-1193 - failed/checkDetails and CI added to VC

### PR DESCRIPTION
**Pull request by ElliotMurphyGDS**
### What changed

- Added check/failed check details functionality to VC
- Added contra Indicator logic to VCs
- added environment variable and permissions to template to allow contra indicator parameter get
- Updated tests

### CODE REVIEW

File: lambdas/issue-credential/src/issue-credential-handler.ts [hunk 1]
```diff
@@ -41,6 +43,7 @@ export class IssueCredentialHandler implements LambdaInterface {
       throw new Error(`Error saving questions to dynamoDb ${errorText}`);
     }
     const correlationId = answerResults.Item.correlationId;
+    logger.info("answerResults: " + JSON.stringify(answerResults));
```
Comment (gds-jbentham) [inline lambdas/issue-credential/src/issue-credential-handler.ts:46]: We probably don't want to log answer results. Same for the whole event above

File: lambdas/issue-credential/src/utils/evidence-builder.ts [hunk 2]
```diff
@@ -29,11 +42,39 @@ export class EvidenceBuilder {
     this.txn = txn;
     return this;
   }
+  addCi(ci: Array<string>) {
+    this.ci = ci;
+    return this;
+  }
+
+  evaluateCheckDetails(checkDetailsLength: number, evidence: Evidence) {
```
Comment (gds-jbentham) [inline lambdas/issue-credential/src/utils/evidence-builder.ts:50]: Why do we need add and evaluate?

File: lambdas/issue-credential/tests/issue-credential-handler.test.ts [hunk 3]
```diff
@@ -71,6 +72,59 @@ const testAnswerResultHappy = {
...
+      {
+        questionKey: "rti-payslip-income-tax",
+        status: "incorrect",
+      },
+      {
+        questionKey: "sa-income-from-pensions",
+        status: "incorrect",
+      },
+    ],
+    verificationScore: 2,
```
Comment (gds-jbentham) [inline lambdas/issue-credential/tests/issue-credential-handler.test.ts:99]: If all questions are incorrect should verification score not be 0

File: lambdas/issue-credential/tests/issue-credential-handler.test.ts [hunk 4]
```diff
@@ -154,4 +222,160 @@ describe("IssueCredentialHandler", () => {
...
+      resultsRetrievalService
+    );
+
+    const sub = "urn:uuid:" + uuidv4().toString();
+    const nbf = Date.now();
+    const iss = mockInputEvent.vcIssuer;
+    const jti = "urn:uuid:" + uuidv4().toString();
+
+    const credentialSubject: CredentialSubject = credentialSubjectBuilder
+      .setSocialSecurityRecord(
```
Comment (gds-jbentham) [inline lambdas/issue-credential/tests/issue-credential-handler.test.ts:246]: Can these common segments be in a util method

File: lambdas/issue-credential/src/issue-credential-handler.ts (hunk 1 above)
Comment (ElliotMurphyGDS) [reply lambdas/issue-credential/src/issue-credential-handler.ts:46]: This was added for debugging forgot to remove

File: package.json [hunk 5]
```diff
@@ -19,11 +19,13 @@
     "@aws-lambda-powertools/commons": "1.14.2",
     "@aws-lambda-powertools/logger": "1.14.2",
     "@aws-lambda-powertools/metrics": "1.14.2",
+    "@aws-lambda-powertools/parameters": "^1.17.0",
     "@aws-lambda-powertools/tracer": "1.14.2",
     "@aws-sdk/client-dynamodb": "3.602.0",
+    "@aws-sdk/client-kms": "3.451.0",
+    "@aws-sdk/client-ssm": "^3.616.0",
```
Comment (gds-jbentham) [inline package.json:26]: Can we remove these to hats

File: lambdas/issue-credential/tests/issue-credential-handler.test.ts (hunk 3 above)
Comment (ElliotMurphyGDS) [reply lambdas/issue-credential/tests/issue-credential-handler.test.ts:99]: Yeah your right good spot

**Comment by ElliotMurphyGDS** [comment]
ContraIndicator parameter will need added to each environment before merging this

**Review by ChrisBates1 — DISMISSED** [review]
Approved from QA. FE Tests conducted to test for chailed/check details and CI values in VC. See Jira Ticket for Test Evidence

File: lambdas/issue-credential/src/utils/check-details-builder.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  kbvQuality: number;
+  checkMethod: string;
+};
+
+export class CheckDetailsBuilder {
+  public buildCheckDetails(checkDetailsCount: number): Array<Check> {
+    const checkDetails: Array<Check> = [];
+    for (let i = 0; i < checkDetailsCount; i++) {
+      checkDetails.push({
+        kbvResponseMode: "free_text",
```
Comment (gds-jbentham) [inline lambdas/issue-credential/src/utils/check-details-builder.ts:12]: Could we make these constants so its clear there hard coded

File: lambdas/submit-answer/src/submit-answer-handler.ts [hunk 7]
```diff
@@ -50,12 +52,17 @@ export class SubmitAnswerHandler implements LambdaInterface {
...
+
       await this.resultService.saveResults(
         event.sessionId,
         event.dynamoResult.Item.correlationId.S,
-        event.usersQuestions.Items[0].expiryDate,
+        ttl,
         answerResult,
-        verificationScore
+        verificationScore,
+        checkDetailsCountCalculator.calculateCorrectAnswers(answerResult),
```
Comment (gds-jbentham) [inline lambdas/submit-answer/src/submit-answer-handler.ts:64]: Can we pull these to variables so if anything fails its easier to identify which line

Comment (gds-jbentham) [reply lambdas/submit-answer/src/submit-answer-handler.ts:64]: Like we have for ttl

File: lambdas/submit-answer/src/utils/check-details-count-calculator.ts [hunk 8]
```diff
@@ -0,0 +1,44 @@
+import { Logger } from "@aws-lambda-powertools/logger";
+import { SubmitAnswerResult } from "../types/answer-result-types";
+
+const logger = new Logger({ serviceName: "CheckDetailsCountCalculator" });
+
+export class CheckDetailsCountCalculator {
+  public calculateCorrectAnswers(answerResults: SubmitAnswerResult[]): number {
```
Comment (gds-jbentham) [inline lambdas/submit-answer/src/utils/check-details-count-calculator.ts:7]: These methods could be one countAnswers and pass it a param of type for correct or incorrect

**Review by gds-jbentham — APPROVED** [review]

**Pull request merged**